### PR TITLE
Slideshow media bundle

### PR DESCRIPTION
**Pull request by vijaynandwani**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,7 +19,21 @@ After downloading the module in the modules folder, visit the Extend (/admin/mod
...
-@ToDo
+1) Enable the module
+2) visit /media/add to add media content to website.
+3) Visit /admin/structure/media to edit the default media bundles.
+
+### Using Slick to modify gallery display
+
+You can use the slick library to improve the gallery display.
+
+1) Download [Slick Library](https://github.com/kenwheeler/slick/) and extract it in libraries directory. Required assets should be available at /libraries/slick/slick/slick.css
```
Comment (slashrsm) [inline README.md]: It would be great to include this in the module by default.

File: config/install/field.field.media.gallery.field_slide.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    - field.storage.media.field_slide
+    - media_entity.bundle.gallery
+    - media_entity.bundle.image
+    - media_entity.bundle.video
+id: media.gallery.field_slide
+field_name: field_slide
+entity_type: media
+bundle: gallery
+label: Slide
+description: ''
```
Comment (slashrsm) [inline config/install/field.field.media.gallery.field_slide.yml]: Is "Slide" really correct label? There are many slides in this field. We should also add a description.

File: src/Tests/GalleryBundleTest.php [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+   * @var \Drupal\media_entity\MediaBundleInterface
+   */
+  protected $testBundle;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    $this->testBundle = $this->container->get('entity.manager')->getStorage('media_bundle')->load('gallery');
```
Comment (slashrsm) [inline src/Tests/GalleryBundleTest.php]: Entity manager is deprecated. Use entity type manager instead.

File: src/Tests/GalleryBundleTest.php [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+      'field_slide[0][target_id]' => $imageItem['name[0][value]'] . ' (' . $imageItem['id'] . ')',
+      'field_slide[0][_weight]' => 0,
+    ];
+    $this->drupalPostAjaxForm(NULL, $edit, 'field_slide_add_more');
+    $edit['field_slide[1][target_id]'] = $videoItem['name[0][value]'] . ' (' . $videoItem['id'] . ')';
+    $edit['field_slide[1][_weight]'] = 1;
+    $this->drupalPostForm(NULL, $edit, t('Save and publish'));
+
+    // Let's load all the media items.
+    $gallery_id = \Drupal::entityQuery('media')->condition('bundle', 'gallery')->sort('created', 'DESC')->execute();
```
Comment (slashrsm) [inline src/Tests/GalleryBundleTest.php]: $this->container instead of \Drupal

File: src/Tests/GalleryBundleTest.php [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+      'field_slide[0][target_id]' => $videoItem['name[0][value]'] . ' (' . $videoItem['id'] . ')',
+      'field_slide[0][_weight]' => 0,
+    ];
+    $this->drupalPostAjaxForm(NULL, $edit, 'field_slide_add_more');
+    $edit['field_slide[1][target_id]'] = $imageItem['name[0][value]'] . ' (' . $imageItem['id'] . ')';
+    $edit['field_slide[1][_weight]'] = 1;
+    $this->drupalPostForm(NULL, $edit, t('Save and publish'));
+
+    // Let's check the thumbnail again.
+    $gallery_id = \Drupal::entityQuery('media')->condition('bundle', 'gallery')->sort('created', 'DESC')->execute();
```
Comment (slashrsm) [inline src/Tests/GalleryBundleTest.php]: Same as above.

File: src/Tests/GalleryBundleTest.php [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+    $file_path = $this->container->get('file_system')->realpath($testImage->uri);
+    $edit = [
+      'name[0][value]' => $name,
+      'files[field_image_0]' => $file_path,
+    ];
+    // Save the image.
+    $this->drupalPostForm('media/add/image', $edit, t('Save and publish'));
+    $this->drupalPostForm(NULL, ['field_image[0][alt]' => $name], t('Save and publish'));
+    // Obtain the image id.
+    $media_id = \Drupal::entityQuery('media')->condition('bundle', 'image')->sort('created', 'DESC')->execute();
```
Comment (slashrsm) [inline src/Tests/GalleryBundleTest.php]: Same as above.

File: src/Tests/GalleryBundleTest.php [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+   * Adds video type item.
+   */
+  public function addVideoItem() {
+    $edit = [
+      'name[0][value]' => 'Drupal video!',
+      'field_video[0][value]' => 'https://www.youtube.com/watch?v=XgYu7-DQjDQ',
+    ];
+    $this->drupalPostForm('media/add/video', $edit, t('Save and publish'));
+    // Obtain the video id.
+    $media_id = \Drupal::entityQuery('media')->condition('bundle', 'video')->sort('created', 'DESC')->execute();
```
Comment (slashrsm) [inline src/Tests/GalleryBundleTest.php]: Same as above.

File: README.md (hunk 1 above)
Comment (vijaynandwani) [inline README.md]: Okay. I'll mention the required libraries as a requirement for the module. And should we include the slick extras module too or mention in the readme for extra styles. https://www.drupal.org/project/slick_extras

Comment (paranojik) [inline README.md]: I would provide a minimal working configuration with as little dependencies as possible (probably just the slick and slick_media modules) and export that.

File: config/install/core.entity_view_display.media.gallery.default.yml [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+  field_slide:
+    weight: 0
+    label: hidden
+    settings:
+      vanilla: true
+      optionset: default
+      optionset_thumbnail: ''
+      skin: ''
+      skin_thumbnail: ''
+      image_style: slick_media
```
Comment (paranojik) [inline config/install/core.entity_view_display.media.gallery.default.yml]: Do we have to call this image style "slick_media". I would expect something that would suggest a gallery item.

**Comment by paranojik** [comment]
I'm afraid you'll have to take care of installing all the dependencies in the 'before' script othwerwise the test won't pass, because media won't install.

File: config/install/core.entity_view_display.media.gallery.default.yml (hunk 8 above)
Comment (paranojik) [inline config/install/core.entity_view_display.media.gallery.default.yml]: Also provide a new "Gallery item" View mode for both "Image" and "Video" and configure it to only display the media item with our image style.

**Comment by vijaynandwani** [comment]
@paranojik dependencies are being installed. Tests are passing. There is still some problem with schema defined incorrectly but I have already opened an issue for that.

**Comment by vijaynandwani** [comment]
Even though I have added our own image style, it is not being used because of the vanilla value being true. You should review it once again to see if we actually need our own image style or not.

**Comment by paranojik** [comment]
what vanilla value are you referring to? we do need our own style because
if someone wants to swap slick for something else the style has to
persist...

**Comment by vijaynandwani** [comment]
@paranojik Yeah that's right. I have made new image style without any effect for gallery items. You can review it now.

File: README.md [hunk 9]
```diff
@@ -19,7 +21,9 @@ After downloading the module in the modules folder, visit the Extend (/admin/mod
 
 ## Configuration
 
-@ToDo
+1. Enable the module
+2. visit /media/add to add media content to website.
+3. Visit /admin/structure/media to edit the default media bundles.
```
Comment (paranojik) [inline README.md:27]: You forgot to transfer the installation instructions from the Media library PR.

Comment (vijaynandwani) [inline README.md:27]: No i didn't. We have included slick module in dependency so those instructions are useless now. I'll remove them from media library PR

**Pull request merged**